src/controller/java/src/chip/devicecontroller/ControllerParams.java

```diff
@@ -151,9 +151,6 @@ public static class Builder {
     private Builder() {}
 
     public Builder setFabricId(long fabricId) {
-      if (fabricId < 1) {
-        throw new IllegalArgumentException("fabricId must be > 0");
-      }
       this.fabricId = fabricId;
       return this;
     }
```

Comment:
I would prefer creating another function
public Builder setFabricId(BigInteger fabricId) {
  return setFabricId(fabricId. longValue())
}

Reply:
@yunhanw-google 
Finally I agree with you. However, since NodeID and FabricID variables are all implemented in long type in the current Java code, shouldn't this code be modified first and then changed overall?

Reply:
Agree. let's remove the restriction first, thanks

Reply:
I think in the follow-up, we just need to setup the convenient setter with BigInteger, so application developer can leverage this setter or implement their conversion code in their side.

Reply:
I understand your comment.